{
  "threat_severity": "Low",
  "public_date": "2018-06-27T00:00:00Z",
  "bugzilla": {
    "description": "CVE-2018-12536 jetty: full server path revealed when using the default Error Handling",
    "id": "1597418",
    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1597418"
  },
  "cvss3": {
    "cvss3_base_score": "3.7",
    "cvss3_scoring_vector": "CVSS:3.0/AV:N/AC:H/PR:N/UI:N/S:U/C:L/I:N/A:N",
    "status": "draft"
  },
  "cwe": "CWE-209",
  "details": [
    "In Eclipse Jetty Server, all 9.x versions, on webapps deployed using default Error Handling, when an intentionally bad query arrives that doesn't match a dynamic url-pattern, and is eventually handled by the DefaultServlet's static file serving, the bad characters can trigger a java.nio.file.InvalidPathException which includes the full path to the base resource directory that the DefaultServlet and/or webapp is using. If this InvalidPathException is then handled by the default Error Handler, the InvalidPathException message is included in the error response, revealing the full server path to the requesting system."
  ],
  "package_state": [
    {
      "product_name": "Red Hat Enterprise Linux 6",
      "fix_state": "Not affected",
      "package_name": "jetty-eclipse",
      "cpe": "cpe:/o:redhat:enterprise_linux:6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Will not fix",
      "package_name": "jetty",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat JBoss Fuse 6",
      "fix_state": "Will not fix",
      "package_name": "jetty",
      "cpe": "cpe:/a:redhat:jboss_fuse:6"
    },
    {
      "product_name": "Red Hat JBoss Fuse 7",
      "fix_state": "Affected",
      "package_name": "jetty",
      "cpe": "cpe:/a:redhat:jboss_fuse:7"
    },
    {
      "product_name": "Red Hat Satellite 5",
      "fix_state": "Will not fix",
      "package_name": "jetty",
      "cpe": "cpe:/a:redhat:network_satellite:5"
    },
    {
      "product_name": "Red Hat Software Collections",
      "fix_state": "Will not fix",
      "package_name": "rh-java-common-jetty",
      "cpe": "cpe:/a:redhat:rhel_software_collections:3"
    }
  ],
  "upstream_fix": "jetty 9.3.24.v20180605,  jetty 9.4.11.v20180605",
  "name": "CVE-2018-12536",
  "mitigation": {
    "value": "Information disclosure occurs when java.nio.file.InvalidPathException occurs and is handled by the default Jetty error handler. To protect against this, a custom error handler can be configured for that particular error or for a larger set of errors according to the documentation link below. Red Hat Product Security advises that production deployments on Jetty use custom error handlers to limit the information disclosed and to ensure effective logging of error conditions.\nhttp://www.eclipse.org/jetty/documentation/current/custom-error-pages.html",
    "lang": "en:us"
  }
}
